packages/destination-actions/src/destinations/snap-conversions-api/reportConversionEvent/index.ts

```
@@ -76,6 +97,22 @@ const action: ActionDefinition<Settings, Payload> = {
       )
     }
 
+    if (data.payload.country && !COUNTRY_ISO_3166_CODES.has(data.payload.country)) {
+      throw new IntegrationError(
```

Hi @sebashine-branly can we throw a PayloadValidationError error instead please?

---

Same concern regarding the fact this field is not required, contrary to what is assumed by the comment above `PayloadValidationError` definition.